compose.yml

```diff
@@ -1,20 +1,73 @@
+# Docker compose file that allows each of the services we require to start
+# only when their dependent service(s) have started. This is achieved through
+# the use of a docker healthcheck for each service.
+
+# First, construct private network for our services to communicate with each other
+networks:
+  network:
+    driver: bridge
+
 services:
-  neo4j-dbms:
-    build: "data/"
+  # neo4j service
+  neo4j-db:
+    image: neo4j:latest
+    environment:
+      NEO4J_AUTH: neo4j/6ti_qvusR!p37WwszTLvzrKbR
+    networks:
+      - network
+    # network ports, we expose non-standard ports so that we
+    # don't conflict with other neo4j instances that may be running
     ports:
-      - 7474:7474
-      - 7687:7687
-    env_file:
-      - "data/.env"
+      - 8474:7474 # HTTP for Neo4j Browser
+      - 8687:7687 # Bolt port
+    expose:
+      - 8474
+      - 8687
+    healthcheck:
+      test: wget http://neo4j-db:7474 || exit 1
+      interval: 5s
+      timeout: 10s
+      retries: 5
+      start_period: 3s
```

Comment:
I think this is something we need to experiment with, I set it to a short value when I was testing. I will change to 30 seconds, but, the knockon effect is that startup may take longer as *I think* it waits that long before its first check.

Reply:
You're right - it does slow down start-up. That is odd - I expected it would do it's 1st health-check but then wait 30 seconds for the next one. As a result, ignore this comment.